htlcswitch/hop/payload.go

```diff
@@ -53,15 +88,24 @@ func NewPayloadFromReader(r io.Reader) (*Payload, error) {
 		return nil, err
 	}
 
-	_, err = tlvStream.DecodeWithParsedTypes(r)
+	parsedTypes, err := tlvStream.DecodeWithParsedTypes(r)
+	if err != nil {
+		return nil, err
```

Comment:
Should the required error also be mapped onto the fancy error type?

Reply:
very likely yes, was thinking we might do that when adding the invalid onion payload failure in the follow up